component/src/main/java/org/wso2/carbon/identity/authenticator/office365/Office365Authenticator.java

```diff
@@ -179,6 +181,30 @@ protected String getLoginType(HttpServletRequest request) {
         }
     }
 
+    /**
+     * Handle error response when unauthorized the registered app.
+     *
+     * @param request httpServletRequest
+     * @throws InvalidCredentialsException
+     */
+    private void handleErrorResponse(HttpServletRequest request) throws InvalidCredentialsException {
+        if (request.getParameter(Office365AuthenticatorConstants.OAUTH2_PARAM_ERROR) != null) {
+            StringBuilder errorMessage = new StringBuilder();
+            String error = request.getParameter(Office365AuthenticatorConstants.OAUTH2_PARAM_ERROR);
+            String errorDescription = request.getParameter
+                    (Office365AuthenticatorConstants.OAUTH2_PARAM_ERROR_DESCRIPTION);
+            String state = request.getParameter(Office365AuthenticatorConstants.STATE);
+            errorMessage.append(Office365AuthenticatorConstants.ERROR).append(error)
+                    .append(Office365AuthenticatorConstants.ERROR_DESCRIPTION).append(errorDescription)
+                    .append(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE).append(state);
+            if (log.isDebugEnabled()) {
+                log.debug("Failed to authenticate via office365 when unauthorized the registered app. " +
```

Comment:
Why we are logging this with debug level? Are we logging this error somewhere else as well?

Reply:
This checking will happened only when we unauthorized the registered app mistakenly or any other specific purposes. So this Debug log is used to make the user get to know the reason behind this error response and this is the only place we are logging this error.

Reply:
In that case shouldn;t this be a error log? Other wise user should enable debug log, to figure out why it is happending.

Reply:
When clicking the "NotNow", from office 365 to unauthorized the user, it will be redirected to IS with following error message and other details.
error : access_denied
error_description : Permissions error
Hence default error message will be showing in the authentication endpoint UI to the user after redirected to IS from office 365.
Do we really need to add error log here?